apps/mobile/src/screens/NftDetailScreen/NftDetailAsset/NftDetailAsset.tsx

```diff
@@ -0,0 +1,170 @@
+import { NftDetailAssetHtml } from '~/screens/NftDetailScreen/NftDetailAsset/NftDetailAssetHtml';
+import { NftDetailAssetImage } from '~/screens/NftDetailScreen/NftDetailAsset/NftDetailAssetImage';
+import { NftDetailAssetVideo } from '~/screens/NftDetailScreen/NftDetailAsset/NftDetailAssetVideo';
+import { Dimensions } from '~/screens/NftDetailScreen/NftDetailAsset/types';
+import { CouldNotRenderNftError } from '~/shared/errors/CouldNotRenderNftError';
+
+type ImageState = { kind: 'loading' } | { kind: 'loaded'; dimensions: Dimensions | null };
+
+type NftDetailProps = {
+  tokenRef: NftDetailAssetFragment$key;
+  style?: ViewProps['style'];
+};
+
+export function NftDetailAsset({ tokenRef, style }: NftDetailProps) {
+  const token = useFragment(
+    graphql`
+      fragment NftDetailAssetFragment on Token {
+        __typename
+
+        media {
+          __typename
+
+          ... on AudioMedia {
+            previewURLs {
+              large
+            }
+          }
+
+          ... on VideoMedia {
+            __typename
+            contentRenderURLs {
+              large
+            }
+          }
+          ... on GIFMedia {
+            __typename
+            previewURLs {
+              large
+            }
+          }
+          ... on ImageMedia {
+            __typename
+            previewURLs {
+              large
+            }
+          }
+
+          ... on HtmlMedia {
+            contentRenderURL
+          }
+        }
+      }
+    `,
+    tokenRef
+  );
+
+  const windowDimensions = useWindowDimensions();
+
+  const [imageState, setImageState] = useState<ImageState>({ kind: 'loading' });
+  const [viewDimensions, setViewDimensions] = useState<Dimensions | null>(null);
+
+  const handleViewLayout = useCallback((event: LayoutChangeEvent) => {
+    const { width, height } = event.nativeEvent.layout;
+
+    setViewDimensions({ width, height });
+  }, []);
+
+  const handleLoad = useCallback((dimensions?: Dimensions | null) => {
+    setImageState({ kind: 'loaded', dimensions: dimensions ?? null });
+  }, []);
+
+  const finalAssetDimensions = useMemo((): Dimensions => {
+    if (viewDimensions && imageState.kind === 'loaded' && imageState.dimensions) {
+      // Give the piece a little bit of breathing room. This might be an issue
+      // if we evers upport landscape view (turning your phone horizontally).
+      const MAX_HEIGHT = windowDimensions.height - 200;
+
+      // Width is the width of the parent view (the screen - some padding)
+      // Height is the max height for the image
+      //
+      // This will fit the image to the screen appropriately.
+      const containerDimensions: Dimensions = { width: viewDimensions.width, height: MAX_HEIGHT };
+
+      return fitDimensionsToContainer({
+        container: containerDimensions,
+        source: imageState.dimensions,
+      });
+    }
+
+    return { width: 300, height: 300 };
```

Comment:
maybe just throw this away

Reply:
this may need to be dynamic if the image is displayed within a smaller container right?